File: backend/xingyun-basedata/src/main/java/com/lframework/xingyun/basedata/vo/product/info/QueryProductSelectorVo.java
package com.lframework.xingyun.basedata.vo.product.info;

import com.lframework.starter.web.core.components.validation.IsEnum;
import com.lframework.starter.web.core.vo.BaseVo;
import com.lframework.starter.web.core.vo.PageVo;
import io.swagger.annotations.ApiModelProperty;
import java.io.Serializable;
import java.time.LocalDateTime;
import lombok.Data;

@Data
public class QueryProductSelectorVo extends PageVo implements BaseVo, Serializable {

  private static final long serialVersionUID = 1L;

  /**
   * 关键字（按编号/名称/SKU模糊匹配）
   */
  @ApiModelProperty("关键字")
  private String condition;

  /**
   * 编号
   */
  @ApiModelProperty("编号")
  private String code;

  /**
   * 名称
   */
  @ApiModelProperty("名称")
  private String name;

  /**
   * SKU
   */
  @ApiModelProperty("SKU")
  private String skuCode;

  /**
   * 简称
   */
  @ApiModelProperty("简称")
  private String shortName;

  /**
   * 品牌ID
   */
  @ApiModelProperty("品牌ID")
  private String brandId;

  /**
   * 分类ID
   */
  @ApiModelProperty("分类ID")
  private String categoryId;

  /**
   * 创建起始时间
   */
  @ApiModelProperty("创建起始时间")
  private LocalDateTime startTime;

  /**
   * 创建截止时间
   */
  @ApiModelProperty("创建截止时间")
  private LocalDateTime endTime;

}

File: frontend/src/api/base-data/product/info/model/queryProductSelectorVo.ts
import { PageVo } from '@/api/model/pageVo';

export interface QueryProductSelectorVo extends PageVo {
  /**
   * 报价单 ID，后端据此排除已有明细商品。
   */
  quoteSheetId?: string;

  /**
   * 关键字
   */
  condition: string;

  /**
   * 编号
   */
  code: string;

  /**
   * 名称
   */
  name: string;

  /**
   * SKU
   */
  skuCode: string;

  /**
   * 简称
   */
  shortName: string;

  /**
   * 品牌ID
   */
  brandId: string;

  /**
   * 分类ID
   */
  categoryId: string;

  /**
   * 创建起始时间
   */
  startTime: string;

  /**
   * 创建截止时间
   */
  endTime: string;

  /**
   * 商品类型
   */
  productType: number;

  /**
   * 状态
   */
  available: boolean;
}
